src/platform_impl/macos/view.rs

```diff
@@ -315,10 +316,17 @@ declare_class!(
 
             // Empty string basically means that there's no preedit, so indicate that by sending
             // `None` cursor range.
+            let mut lowerbound: usize = preedit_string.len();
+            let mut upperbound: usize = lowerbound;
+            if let Some(fetched_value) = get_actual_range_bounds(&preedit_string, selected_range) {
+                lowerbound = fetched_value.0;
+                upperbound = fetched_value.1;
+            }
```

Comment:
I would prefer this solution:

```suggestion
            let (lowerbound, upperbound) = if let Some(fetched_value) = get_actual_range_bounds(&preedit_string, selected_range).unwrap_or_else(|| (preedit_string.len(), lowerbound));
```

I did not test if this passes the compilation.

Reply:
Thanks for your suggestion. `.unwrap_or(fallbackValue)` is enough in this case.